src/esp/nav/PathFinder.cpp

```diff
@@ -885,78 +950,35 @@ bool PathFinder::Impl::findPath(MultiGoalShortestPath& path) {
   }
 
   std::vector<vec3f> pathEnds;
-  std::vector<float> pathEndsCoords;
   std::vector<dtPolyRef> endRefs;
   for (const auto& rqEnd : path.requestedEnds) {
     pathEnds.emplace_back();
     endRefs.emplace_back();
     std::tie(status, endRefs.back(), pathEnds.back()) =
         projectToPoly(rqEnd, navQuery_.get(), filter_.get());
 
-    pathEndsCoords.emplace_back(pathEnds.back()[0]);
-    pathEndsCoords.emplace_back(pathEnds.back()[1]);
-    pathEndsCoords.emplace_back(pathEnds.back()[2]);
-
     if (status != DT_SUCCESS || endRefs.back() == 0) {
       return false;
     }
   }
 
-  // check if trivial path (start is same as end) and early return
-  if (std::find_if(path.requestedEnds.begin(), path.requestedEnds.end(),
-                   [&path](const vec3f& rqEnd) -> bool {
-                     return rqEnd.isApprox(path.requestedStart);
-                   }) != path.requestedEnds.end()) {
-    path.geodesicDistance = 0;
-    return true;
-  }
-
-  // Check if there is a path between the start and any of the ends
-  if (std::find_if(endRefs.begin(), endRefs.end(),
-                   [this, &startRef](const dtPolyRef& end) -> bool {
-                     return this->islandSystem_->hasConnection(startRef, end);
-                   }) == endRefs.end()) {
-    return false;
-  }
-
-  int goalFoundIdx;
-  status = navQuery_->findBidirPathToAny(
-      endRefs.size(), startRef, endRefs.data(), path.requestedStart.data(),
-      pathEndsCoords.data(), filter_.get(), polys, &numPolys, MAX_POLYS,
-      &goalFoundIdx);
-  if (status != DT_SUCCESS) {
-    return false;
-  }
-
-  if (numPolys) {
-    const vec3f& closestRequestedEnd = path.requestedEnds[goalFoundIdx];
-
-    path.points.resize(MAX_POLYS);
-    status = navQuery_->findStraightPath(
-        path.requestedStart.data(), closestRequestedEnd.data(), polys, numPolys,
-        path.points[0].data(), 0, 0, &numPoints, MAX_POLYS);
+  for (int i = 0; i < path.requestedEnds.size(); ++i) {
+    if ((path.requestedStart - path.requestedEnds[i]).norm() >
```

Comment:
As I understand this check is selecting prospect end points that can be closer that `path.geodesicDistance`

Reply:
Yep, exactly.